packages/harmony/src/components/input/TextInput/TextInput.tsx

```diff
@@ -0,0 +1,213 @@
+import { forwardRef, type ComponentPropsWithoutRef } from 'react'
+
+import cn from 'classnames'
+
+import layoutStyles from 'components/layout/layout.module.css'
+import { IconComponent, Text } from 'components/typography'
+
+import { HelperText } from './HelperText'
+import styles from './TextInput.module.css'
+import { useFocusState } from './useFocusState'
+
+export enum TextInputSize {
+  SMALL = 'small',
+  DEFAULT = 'default'
+}
+
+export type TextInputProps = Omit<
+  ComponentPropsWithoutRef<'input'>,
+  // Omitting required purely for storybook docs
+  'size' | 'required'
+> & {
+  /**
+   * Input sizes. NOTE: small inputs will not show the label
+   * @default default
+   */
+  size?: TextInputSize
+  /**
+   * Show a "x/Max" text displaying number of characters allowed
+   */
+  showMaxLength?: boolean
```

Comment:
I remember design had a request to be able to show this conditionally when you get close to the end. Don't need to add scope here, but flagging anyway

Reply:
I met up with Sammie and re-did this, now it has a threshold for showing and a threshold for changing to the warning color